java-compute/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITSmokeInstancesTest.java

```diff
@@ -72,6 +72,8 @@ public static void setUp() throws IOException {
     instances = new ArrayList<>();
     InstancesSettings instanceSettings = InstancesSettings.newBuilder().build();
     instancesClient = InstancesClient.create(instanceSettings);
+
+    Util.cleanUpComputeInstances(instancesClient, DEFAULT_PROJECT, DEFAULT_ZONE);
```

Comment:
Isn't this a bit too aggressive to delete all compute instances? Why isn't the cleanup in `tearDown()` sufficient?

Reply:
tearDown() cleanup is sufficient for 99% of the case. There was a significant amount of compute instances that were in there (my guess from previous runs that didn't have the cleanup or when the IT jobs were cancelled halfway). 

This project is only used for ITs so it would have 24 hours to report the results before the previous instances are deleted.

Reply:
I'm worried that someone runs the test on their own project and causes inadvertent deletion of instances. If we want to do this kind of cleanup, maybe we can have instance name prefix that is consistent and can accurately filter instances created by this test.

Reply:
> I'm worried that someone runs the test on their own project and causes inadvertent deletion of instances.

Actually, I was thinking that this would be fine for the ITs. I was thinking the ideal state for gcloud-devel would have 0 compute instances when there are no active ITs running. If `Util.cleanUpComputeInstances` (or more filtered version of that function) matches any existing instances, then those should be deleted. 24 hours is 12x as long as the longest GraalVM job.

> maybe we can have instance name prefix that is consistent and can accurately filter instances created by this test. 

Yep! I was planning on creating another PR that would update the instances for each module to have a more unique name instead `test-instance` or `gapic-instance`. i.e. Compute would be test-compute-instance-UUID, Container would be test-container-instance-UUID or something like that.

Reply:
Ok nvm, I just re-read the comment. 

>  test on their own project 
That makes more sense.